packages/client/hmi-client/src/services/model-configurations.ts

```diff
@@ -67,3 +74,75 @@ export const updateModelConfiguration = async (config: ModelConfiguration) => {
 	const response = await API.put(`/model-configurations/${config.id}`, config);
 	return response?.data ?? null;
 };
+
+export const getInitial = (config: ModelConfiguration, initialId: string): Initial | undefined =>
+	config.configuration.semantics?.ode.initials?.find((initial) => initial.target === initialId);
+
+export const getInitialSource = (config: ModelConfiguration, initialId: string): string =>
+	config.configuration.metadata?.initials?.[initialId].source ?? '';
+
+export const setInitialSource = (
+	config: ModelConfiguration,
+	initialId: string,
+	source: string
+): void => {
+	const initial = config.configuration.metadata?.initials?.[initialId];
+	if (initial) {
+		initial.source = source;
+	}
+};
+
+export const getParameter = (
+	config: ModelConfiguration,
+	parameterId: string
+): ModelParameter | undefined =>
+	config.configuration.semantics?.ode.parameters?.find((param) => param.id === parameterId);
+
+export const setDistribution = (
+	config: ModelConfiguration,
+	distribution: ModelDistribution,
+	parameterId: string
+): void => {
+	const parameter = getParameter(config, parameterId);
+	if (parameter) {
+		parameter.distribution = distribution;
+	}
+};
+
+export const removeDistribution = (config: ModelConfiguration, parameterId: string): void => {
+	const parameter = getParameter(config, parameterId);
+	if (parameter?.distribution) {
+		delete parameter.distribution;
+	}
+};
+
+export const getInterventions = (config: ModelConfiguration): Intervention[] =>
+	config.interventions ?? [];
+
+export const setIntervention = (
+	config: ModelConfiguration,
+	intervention: Intervention,
+	index: number
+): void => {
+	const interventions = getInterventions(config);
+	interventions[index] = intervention;
+};
+
+export const removeIntervention = (config: ModelConfiguration, index: number): void => {
+	const interventions = getInterventions(config);
+	interventions.splice(index, 1);
+};
```

Comment:
Just a heads up - using index to "set" and "get" is not a good API design, because indices change as the data structure is modified. I don't know if this is meant as a temporary measure or not, if so, please add a a FIXME.

Reply:
definitely temporary for now - ill add a comment